### DISCUSSION TITLE
Some general install issues and solutions

### DISCUSSION THREAD

**Discussion by datasprings** [Leci37/TensorFlow-stocks-prediction-Machine-learning-RealTime#19] [General]
Hello... Recently ran through the install/setup instructions and wanted to post some updates in case others have the same issue. 

1. Biggest issue was with the ta-lib install... I went through many posts and related discussions to address this, but the short answer is that I had to execute the instructions found here: 

https://stackoverflow.com/questions/41155985/python-ta-lib-install-problems (I am running on a 64 bit version of windows and I had to download the VS commandline build and execute the nmake.exe file for it to properly end up working and functioning later). 

https://stackoverflow.com/questions/27885397/how-do-i-install-a-python-package-with-a-whl-file

I believe I also had to download the source from (https://www.lfd.uci.edu/~gohlke/pythonlibs/#ta-lib) and because my Python was newer, I had to rename one of the files so that PIP would not give me an error that it was not supported.

pip debug --verbo -- If the name is not in your list, you will need to rename it to one that is (I guess because I have newest version of Python?) 

Then... 
pip install TA_Lib-0.4.24-cp311-cp311-win_amd64.whl 

2. From the requirements.txt I also had to remove the line related to TensorFlow GPU, and I had to manually do a pip install for:
pip install investpy==0.9.1

3. I had to downgrade the version of https://scikit-learn.org/stable/install.html  to 1.2.2 because of some errors.  

Thanks,

-Chad
www.nashtech.xyz

**Comment by theharold** [comment]
> Hello... Recently ran through the install/setup instructions and wanted to post some updates in case others have the same issue.
> 
> 1. Biggest issue was with the ta-lib install... I went through many posts and related discussions to address this, but the short answer is that I had to execute the instructions found here:
> 
> https://stackoverflow.com/questions/41155985/python-ta-lib-install-problems (I am running on a 64 bit version of windows and I had to download the VS commandline build and execute the nmake.exe file for it to properly end up working and functioning later).
> 
> https://stackoverflow.com/questions/27885397/how-do-i-install-a-python-package-with-a-whl-file
> 
> I believe I also had to download the source from (https://www.lfd.uci.edu/~gohlke/pythonlibs/#ta-lib) and because my Python was newer, I had to rename one of the files so that PIP would not give me an error that it was not supported.
> 
> pip debug --verbo -- If the name is not in your list, you will need to rename it to one that is (I guess because I have newest version of Python?)
> 
> Then... pip install TA_Lib-0.4.24-cp311-cp311-win_amd64.whl
> 
> 2. From the requirements.txt I also had to remove the line related to TensorFlow GPU, and I had to manually do a pip install for:
>    pip install investpy==0.9.1
> 3. I had to downgrade the version of https://scikit-learn.org/stable/install.html  to 1.2.2 because of some errors.
> 
> Thanks,
> 
> -Chad [www.nashtech.xyz](http://www.nashtech.xyz)

Did you see https://github.com/Leci37/stocks-prediction-Machine-learning-RealTime-telegram/blob/master/Readme_setup_installation.md
